drivers/counter/Kconfig.nrfx

```diff
@@ -9,12 +9,6 @@ config COUNTER_NRF_RTC
 	def_bool y
 	depends on DT_HAS_NORDIC_NRF_RTC_ENABLED
 
-config COUNTER_NRFX_TIMER_USE_CLOCK_CONTROL
```

Comment:
The commit message says "adjust USE_CLOCK_CONTROL" but this option is actually removed. I guess the related change from the third commit should be squashed with this one.